greeting.py

```diff
@@ -1,13 +1,44 @@
 def greet(name):
     if name is None:
-        name = "my friend"
-    elif isinstance(name, list):
-        name = f"{name[0]} and {name[1]}"
+        return get_greet_from_string("my friend")
+    if isinstance(name, list):
+        return get_greet_from_list(name)
     else:
-        name = str(name)
+        return get_greet_from_string(str(name))
+
 
+def get_greet_from_string(name):
     if name.isupper():
-        result = f"HELLO {name}!"
+        return f"HELLO {name}!"
+    else:
+        return f"Hello, {name}."
+
+
+def get_greet_from_list(name):
+    normal_name = [n for n in name if not n.isupper()]
+    shouting_name = [n for n in name if n.isupper()]
+    normal_name_count = len(normal_name)
+    shouting_name_count = len(shouting_name)
+    normal_greet = get_greet_from_list_in_one_case(normal_name, 0) if normal_name_count > 0 else ""
```

Comment:
why use integers in place of booleans? also, it's hard to understand what the second parameter means. I would suggest to rename the function and use `kwargs` for `is_shouting`. so it will look like
```suggestion
    normal_greet = get_greet_from_list(normal_name, is_shouting=False) if normal_name else ""
```